qep-400-coding-practice.md

```diff
@@ -0,0 +1,91 @@
+# QGIS Enhancement 400: Code style and practice guidelines
+
+## Summary
+
+This QEP documents coding standards and conventions used throughout the QGIS code base. Developers are required to follow these standards when submitting code to QGIS.
+
+Notes:
+
+- Spacing and indentation rules are automatically applied by the QGIS pre-commit hooks, and accordingly are not covered here
+- These guidelines describe QGIS specific coding practices only. Following "best practice" for modern c++ coding is assumed, and only referred to here when QGIS coding requires certain variances from this.
+
+## Coding Guidelines
+
+### 1. Naming
+
+- 1.1. Variable and function naming should be camel case. Generally Qt's approach to capitalising only the first letter in acronyms is used (eg ``QgsGpsDetector``, not ``QgsGPSDetector``), with the exception of ``3D`` (eg ``Qgs3DMapCanvas``).
+
+  - 1.1.1. Local variables should have no name prefix (eg do not prefix variable names with ``my``)
+  - 1.1.2. [Hungarian notation](https://en.m.wikipedia.org/wiki/Hungarian_notation) is NOT used, except in some exceptional circumstances (eg variable names used with the gdal c API)
+  - 1.1.3. Member variables should have a ``m`` prefix (eg ``mLineWidth``)
```

Comment:
Does it also apply to struct? It doesn't seem consistent over the code base
- [with](https://github.com/qgis/QGIS/blob/64e2a010199db7a655aab70ae968966e550efc4d/src/core/expression/qgsexpression.h#L92)
- [without](https://github.com/qgis/QGIS/blob/64e2a010199db7a655aab70ae968966e550efc4d/src/core/effects/qgsimageoperation.h#L221)

Maybe it's worth mentioning here wether it applies to class AND struct or only to the first one

Reply:
Indeed, there's no consistency here. I'll omit any formal specification for now until we decide on one.